src/js/src/views/SolrWayback.vue

```diff
@@ -41,55 +43,61 @@ export default {
       updateSolrSettingGrouping:'updateSolrSettingGrouping',
       updateSolrSettingImgSearch:'updateSolrSettingImgSearch',
       updateSolrSettingUrlSearch:'updateSolrSettingUrlSearch',
-      updateFutureSolrSettingGrouping:'updateFutureSolrSettingGrouping',
-      updateFutureSolrSettingImgSearch:'updateFutureSolrSettingImgSearch',
-      updateFutureSolrSettingUrlSearch:'updateFutureSolrSettingUrlSearch'
+      updateSolrSettingOffset:'updateSolrSettingOffset',
     }),
     onScroll(e) {
     e.target.documentElement.scrollTop > 0 ? this.scrolledFromTop = true : this.scrolledFromTop = false
     },
     backToTop() {
       window.scrollTo({ top: 0, behavior: 'smooth' })
-    }, 
+    },
   },
   beforeRouteUpdate (to, from, next) {
-    //console.log('route changed!',to)
-    this.updateQuery(to.query.q)
-    to.query.grouping === 'true' ? (this.updateSolrSettingGrouping(true), this.updateFiutureSolrSettingGrouping(true)) : (this.updateSolrSettingGrouping(false), this.updateFutureSolrSettingGrouping(false))
-    to.query.imgSearch === 'true' ? (this.updateSolrSettingImgSearch(true), this.updateFutureSolrSettingImgSearch(true)) : (this.updateSolrSettingImgSearch(false), this.updateFutureSolrSettingImgSearch(false))
-    to.query.urlSearch === 'true' ? (this.updateSolrSettingUrlSearch(true), this.updateFutureSolrSettingUrlSearch(true)) :  (this.updateSolrSettingUrlSearch(false), this.updateFutureSolrSettingUrlSearch(false))
-    to.query.facets ? this.updateSearchAppliedFacets(to.query.facets) : this.updateSearchAppliedFacets('')
-    if(this.solrSettings.imgSearch) {
-      this.requestImageSearch({query:this.query})
+    //console.log('route changed!',to.query, from.query)
+    // Check if any of our params have changed. Could be refactored into a nice functon.
+    if(to.query.query !== undefined &&
+       (to.query.query !== from.query.query ||
+       to.query.offset !== from.query.offset ||
+       to.query.imgSearch !== from.query.imgSerach ||
+       to.query.urlSearch !== from.query.urlSearch ||
+       to.query.grouping !== from.query.grouping ||
+       to.query.facets !== from.query.facets)) {
+      //console.log('we doing a route search')
+      // update our variables from the query.
+      to.query.grouping === 'true' || to.query.grouping === true ? this.updateSolrSettingGrouping(true) : this.updateSolrSettingGrouping(false)
+      to.query.imgSearch === 'true' || to.query.imgSearch === true ? this.updateSolrSettingImgSearch(true) : this.updateSolrSettingImgSearch(false)
+      to.query.urlSearch === 'true' || to.query.urlSearch === true ? this.updateSolrSettingUrlSearch(true) : this.updateSolrSettingUrlSearch(false)
+      to.query.urlSearch === 'true' || to.query.urlSearch === true ? this.updateSolrSettingUrlSearch(true) : this.updateSolrSettingUrlSearch(false)
+      to.query.offset ? this.updateSolrSettingOffset(Number(to.query.offset)) : this.updateSolrSettingOffset(0)
```

Comment:
I see some Number() casting action here - be aware that unchecked this can result in a _NaN_ error

Reply:
Offset should never be anything than a number! But sometimes, again, when the backbutton is used, we might get offset: "0", which we then cast to a number to make sure we always have an integer here. Otherwise the whole trying to add a number when we press button for "next 10" becomes tainted and we end up with "0" + 20, which gives us 020 - you get the idea.

Reply:
If a direct link messes with the offset so it comes in as _undefined_ the code will break. This could happen in a copy/paste error made by end-user  As long as this is accounted for somewhere in the flow I'm good :)